File: Data Structures & Algorithms/minimum-changes-to-make-alternating-binary-string/submission-1.py
class Solution:
    def minOperations(self, s: str) -> int:
        def check(start_char):
            res = 0
            curr = start_char
            for char in s:
                if char != curr:
                    res += 1
                curr = '1' if curr == '0' else '0'
            return res

        return min(check('0'), check('1'))

File: Data Structures & Algorithms/max-water-container/submission-1.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int l = 0;
        int r = heights.size()-1;
        int res = 0;

        while (l < r){
            int height = min(heights[l], heights[r]);
            res = max(res, height * (r-l));

            if (heights[l] > heights[r]) {
                r--;
            } else{
                l++;
            }
        }

        return res;
    }
};
